File: fastjson1-compatible/src/test/java/com/alibaba/fastjson/JSONPathTest.java
package com.alibaba.fastjson;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class JSONPathTest {
    @Test
    public void remove() {
        JSONObject object = new JSONObject().fluentPut("id", 123);
        assertTrue(JSONPath.remove(object, "$.id"));
        assertTrue(object.isEmpty());

        assertTrue(JSONPath.set(object, "$.id", 234));
        assertFalse(object.isEmpty());
        assertEquals(234, object.get("id"));
    }

    @Test
    public void compile() {
        JSONPath path = JSONPath.compile("$.id");
        assertEquals("$.id", path.getPath());

        JSONObject object = new JSONObject().fluentPut("id", 123);
        assertEquals(123, path.eval(object));
        path.set(object, 234);
        assertEquals(234, object.getIntValue("id"));
    }

    @Test
    public void error() {
        assertThrows(Exception.class, () -> JSONPath.compile(null));
    }
}

File: kotlin/src/test/kotlin/com/alibaba/fastjson2/JSONPathTest.kt
package com.alibaba.fastjson2

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class JSONPathTest {
    @Test
    fun test_eval() {
        val user = User(
            1, "kraity"
        )
        val name = user.eval("$.name")

        assertTrue(name is String)
        assertEquals("kraity", name)
    }

    @Test
    fun test_contains() {
        val user = User(
            1, "kraity"
        )

        assertTrue(user.contains("$.id"))
        assertTrue(user.contains("$.name"))
        assertFalse(user.contains("$.blocked"))
    }

    data class User(
        val id: Int,
        val name: String
    )
}
